collection.upload.staticURL is optional but is actually required

### Link to reproduction

https://github.com/payloadcms/payload/commit/cf8b836f72011a80f92eabedb49a35d970652974

### To Reproduce

1. `payload.init({ serverUrl: undefined })`
2. Create collection "test" without upload.staticURL
3. Add a document
4. Query the collection and notice the url property


Option 1: 
Expected: `url: "/test/example.png"`
Actual: `url: "undefined/test/example.png"`

Option 2: 
Expected: invalid payload config, staticURL was empty
Actual: valid payload config, image urls are broken

### Describe the Bug

This worked upto 1.13, it broke it 1.14.0

Ideally it should use the server origin + collection slug (as it did before?)

### Payload Version

1.14 - 1.15.2

OK, here it is: https://github.com/thgh/payload/tree/fix/3237-server-url
Run `yarn test:int -t uploads` and it will succeed, revert the sanitize fix to trigger the bug

Or check out this [action](https://github.com/payloadcms/payload/actions/runs/5998851506/job/16267864144) (see log below)

[(here are some notes on the root cause)](https://github.com/payloadcms/payload/pull/3239#issuecomment-1695816583)

```
Summary of all failing tests
FAIL test/uploads/int.spec.ts (14.899 s)
  ● Collections - Uploads › REST › create › creates from form data given a png

    expect(received).not.toContain(expected) // indexOf

    Expected substring: not "undefined"
    Received string:        "undefined/media/image-1.png"

      58 |         expect(sizes).toHaveProperty('mobile');
      59 |         expect(sizes).toHaveProperty('icon');
    > 60 |         expect(doc.url).not.toContain('undefined');
         |                             ^
      61 |       });
      62 |
      63 |       it('creates from form data given an svg', async () => {

      at Object.toContain (test/uploads/int.spec.ts:60:29)


Test Suites: 1 failed, 24 passed, 25 total
Tests:       1 failed, 1 skipped, 2 todo, 452 passed, 456 total
Snapshots:   0 total
Time:        94.596 s
Ran all test suites.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Process completed with exit code 1.
```

--- a/packages/payload/src/config/sanitize.ts
+++ b/packages/payload/src/config/sanitize.ts
@@ -45,6 +45,10 @@ export const sanitizeConfig = (incomingConfig: Config): SanitizedConfig => {
     isMergeableObject: isPlainObject,
   }) as Config
 
+  if (!configWithDefaults.serverURL) {
+    configWithDefaults.serverURL = ''
+  }
+
   const config: Partial<SanitizedConfig> = sanitizeAdminConfig(configWithDefaults)
 
   if (config.localization && config.localization.locales?.length > 0) {
@@ -90,10 +94,6 @@ export const sanitizeConfig = (incomingConfig: Config): SanitizedConfig => {
 
   if (config.globals.length > 0) {
     config.globals = sanitizeGlobals(config as SanitizedConfig)
-  }
-
-  if (typeof config.serverURL === 'undefined') {
-    config.serverURL = ''
   }
 
   if (config.serverURL !== '') {

--- a/test/uploads/config.ts
+++ b/test/uploads/config.ts
@@ -12,6 +12,7 @@ import { audioSlug, enlargeSlug, mediaSlug, reduceSlug, relationSlug } from './s
 const mockModulePath = path.resolve(__dirname, './mocks/mockFSModule.js')
 
 export default buildConfigWithDefaults({
+  serverURL: undefined,
   admin: {
     webpack: (config) => ({
       ...config,

--- a/test/uploads/int.spec.ts
+++ b/test/uploads/int.spec.ts
@@ -84,6 +84,22 @@ describe('Collections - Uploads', () => {
       })
     })
 
+    it('should have valid image url', async () => {
+      const formData = new FormData()
+      formData.append('file', fs.createReadStream(path.join(__dirname, './image.png')))
+
+      const { status, doc } = await client.create({
+        file: true,
+        data: formData,
+      })
+
+      expect(status).toBe(201)
+      const expectedPath = path.join(__dirname, './media')
+      expect(await fileExists(path.join(expectedPath, doc.filename))).toBe(true)
+
+      expect(doc.url).not.toContain('undefined')
+    })
+
     it('creates images that do not require all sizes', async () => {
       const formData = new FormData()
       formData.append('file', fs.createReadStream(path.join(__dirname, './small.png')))

